main.go

```diff
@@ -80,10 +80,10 @@ func imageFromContext(imageName string, count int) []registry.SearchResult {
 
 func imageFetchCompleter(imageName string, count int) []prompt.Suggest {
 	searchResult := []registry.SearchResult{}
-	if imageName == "" {
-		searchResult = imageFromHubAPI(10)
-	} else {
+	if imageName != "" {
 		searchResult = imageFromContext(imageName, 10)
+	} else {
+		searchResult = imageFromHubAPI(10)
```

Comment:
To prevent slowing down the typing what about fetch the most pulled 10 images on startup in a different go routine? We can also handle timeouts with a retry to fetch images on that goroutine.

Reply:
Totally agree with you! It's implemented.